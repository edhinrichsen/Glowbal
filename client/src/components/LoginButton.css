.ant-btn {
  border-radius: 10px;
  background: var(--btn-gradient) !important;
  border-radius: 100px !important;
  border: var(--border);
  height: 40px; 
  box-shadow: inset 0px 3px 5px rgba(0, 0, 0, 0.1), var(--drop-shadow);
}

.ant-btn:hover {
  border: var(--border);
  box-shadow: 0px 0px 0px 1px var(--color-primary), inset 0px 3px 5px rgba(0, 0, 0, 0.1), var(--drop-shadow);
}
