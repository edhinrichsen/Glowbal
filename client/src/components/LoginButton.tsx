import React from 'react';
import { useAuth0 } from '@auth0/auth0-react';

import { Button } from 'antd';

import './LoginButton.css';

function LoginButton() {
  const { isAuthenticated, loginWithRedirect, logout } = useAuth0();

  return (
    <div>
      {isAuthenticated ? (
        <Button onClick={() => logout({ returnTo: window.location.origin })}>
          Logout
        </Button>
      ) : (
        <Button type="primary" onClick={() => loginWithRedirect()}>
          Login
        </Button>
      )}
    </div>
  );
}

export default LoginButton;
