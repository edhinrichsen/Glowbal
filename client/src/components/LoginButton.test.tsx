import React from 'react';
import { shallow, mount } from 'enzyme';
import LoginButton from './LoginButton';

describe('Testing LoginButton', () => {

  it('renders without crashing', () => {
    const component = shallow(<LoginButton />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('shows login when not authenticated', () => {
    const component = mount(<LoginButton />);

    component.find({ children: 'Login' });
    expect(component).toMatchSnapshot();
    
    component.unmount();
  });

})

