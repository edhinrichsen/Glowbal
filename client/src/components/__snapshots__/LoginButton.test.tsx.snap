// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginButton renders without crashing 1`] = `
<div>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    type="primary"
  >
    Login
  </Button>
</div>
`;

exports[`Testing LoginButton shows login when not authenticated 1`] = `
<LoginButton>
  <div>
    <Button
      block={false}
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[Function]}
      type="primary"
    >
      <Wave>
        <button
          className="ant-btn ant-btn-primary"
          onClick={[Function]}
          type="button"
        >
          <LoadingIcon
            existIcon={false}
            loading={false}
            prefixCls="ant-btn"
          >
            <CSSMotion
              motionName="ant-btn-loading-icon-motion"
              onAppearActive={[Function]}
              onAppearStart={[Function]}
              onEnterActive={[Function]}
              onEnterStart={[Function]}
              onLeaveActive={[Function]}
              onLeaveStart={[Function]}
              removeOnLeave={true}
              visible={false}
            >
              <DomWrapper />
            </CSSMotion>
          </LoadingIcon>
          <span
            key=".0"
          >
            Login
          </span>
        </button>
      </Wave>
    </Button>
  </div>
</LoginButton>
`;
